### ISSUE TITLE
[DeviceMesh][Collective] Support multiple tensors API

### ISSUE THREAD

**Issue by jiaodong** [ray-project/ray#25129]
### Description

Currently apis such as send/recv/broadcast only supports sending one tensor per call, and due to the fact it uses default cuda stream, the call is synchronous. 

But for Ray libraries like RLlib we have data structures where there're multiple tensors involved for each policy, such as 

{
"_logits._model.0.weight": ndarray
"_hidden_layers.1._model.0.weight": ndarray
}

In this case ideally we should be able send all tensors in one batch, rather than sending in a for loop.

### Use case

RLlib on GPU

**Comment by jovany-wang** [comment]
Do you plan to support it for Gloo based collective?

**Comment by jiaodong** [comment]
@jovany-wang I haven't looked into gloo yet but I saw some PRs from you using it. Do you mind sharing what use cases you have and the ideal APIs you have in mind ? I'm only prototyping RLlib on GPUs but curious to learn more.

**Comment by jiaodong** [comment]
keep alive

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
